core/shared/src/main/scala/fs2/pipe.scala

```diff
@@ -135,6 +136,28 @@ object pipe {
     }
   }
 
+  /** Debounce the stream with a minimum delay of `d` between each element */
+  def debounce[F[_], I](d: FiniteDuration)(implicit F: Effect[F], scheduler: Scheduler, ec: ExecutionContext): Pipe[F, I, I] = {
```

Comment:
I'm very sorry, but I'm not too sure how this would work; the difficulty in the debounce seems to me to be in the fact that your `time.sleep` has to from the point of each new `A` which is emitted by `in` _and_ the ability to discard an element if the next one is emitted before the period.

Could you elaborate a little on what you mean?